packages/amplication-server/src/core/auth/auth.service.ts

```diff
@@ -423,6 +527,72 @@ export class AuthService {
     return workspace as unknown as Workspace & { users: AuthUser[] };
   }
 
+  private async createPreviewWorkspaceWithProjectAndResource(account: Account) {
+    const workspaceName = `Amplication-${this.generateRandomString()}`;
+    const workspace = (await this.workspaceService.createPreviewWorkspace(
+      {
+        data: {
+          name: workspaceName,
+        },
+        include: WORKSPACE_INCLUDE,
+      },
+      account.id
+    )) as unknown as Workspace & { users: AuthUser[] };
+    const [user] = workspace.users as AuthUser[];
+
+    const project = await this.projectService.createPreviewProject(
+      {
+        data: {
+          name: account.previewAccountType,
```

Comment:
Do we want to use the same project name for all preview workspaces? (perhaps it's in the spec and I don't remember)

Reply:
Yep. This is based on the spec.